add full image name in overview

shun-miyoshi-com opened the pull request:
This pull request will resolve issue #2286.
I added full image name in overview.
We can see full image name by hovering the mouse over "Images".
But, now I implemented of resource "Deployments" only.

If you accept my this concept, I will implement all of resources.
Thank you.

shun-miyoshi-com commented:
I signed a CLA.

floreks commented:
This is actually what we wanted to avoid. Tooltip should be only displayed in case name is truncated, not always.

shun-miyoshi-com commented:
I changed my code.
My algorithm is shown as follows.

1. get field width.
  I get field width of Images header column.
  * document.getElementById("deployment_image_header").clientWidth

2. get display length of image full name
  I added hidden <div> tag, and get its width
  * <div style="position:absolute;visibility:hidden" ng-attr-id="{{'deployment_image_'+image}}">{{::image}}</div>
  * document.getElementById("deployment_image_"+imageName).clientWidth

3. compare 1. and 2.
  If 2 .> 1. , an image name may be truncated.

Please review my code.

floreks commented:
I don't have time to investigate but this does not work in all cases. Some of my images are truncated but tooltip is not displayed.

You can check out this thread: https://stackoverflow.com/questions/7738117/html-text-overflow-ellipsis-detection

Also I'd preffer to keep the logic inside `kdMiddleEllipsis` component so it would always work.

shun-miyoshi-com commented:
Your suggestion is really interesting and cool.
Unfortunately, I cannot implement this thread's code completely, because angularjs cannot use JQuery(only JQLite).
JQLite cannot use width() method.(please see https://docs.angularjs.org/api/ng/function/angular.element)
So, I use offsetWidth, however this variable don't run by element.clone() method.
Thus, I use addClass()/removeClass() method instead of clone()/remove() method.

maciaszczykm commented on src/app/frontend/common/components/middleellipsis/component.js:
```diff
@@ -19,10 +19,26 @@ export default class MiddleEllipsisController {
+
+  isTrancated() {
+    let c = this.element_;
+    c.addClass("kd-middleellipsis_width");
```
Use Angular's `ng-class`.

shun-miyoshi-com replied:
I'm sorry, but I don't understand you.
What should I do?

maciaszczykm replied:
https://docs.angularjs.org/api/ng/directive/ngClass

shun-miyoshi-com replied:
I have no idea how to use `ng-class` effectively.
The example of the link is how to add class dynamically, but I use addClass method because of adding class temporarily.
Please help me.

maciaszczykm commented on src/app/frontend/common/components/middleellipsis/component.js:
```diff
@@ -19,10 +19,26 @@ export default class MiddleEllipsisController {
+    this.element_ = $element;
+  }
+
+  isTrancated() {
```
Missing annotations. Check our Wiki pages for more info about code conventions.

shun-miyoshi-com replied:
Which annotation is mistaken?(this.element_ or isTrancated or any other else ?)
this.element_ is used in  MiddleEllipsisController only, and isTruncated is used in middleellipsis.html.

maciaszczykm replied:
Closure compiler annotations - read about them in our wiki.

shun-miyoshi-com replied:
My added annotations( @ param {!angular.JQLite} $element and @ private {!angular.JQLite}) will same as other components such as (common/components/allocatedresourceschart/component.js)
I'm sorry, but could you tell me about a detail of mistaken annotetions.

maciaszczykm commented on src/app/frontend/common/components/middleellipsis/component.js:
```diff
@@ -19,10 +19,26 @@ export default class MiddleEllipsisController {
+  isTrancated() {
+    let c = this.element_;
+    c.addClass("kd-middleellipsis_width");
+    let c_ofs_width = c[0].offsetWidth;
```
Use camelCase.

maciaszczykm commented:
Any update?

Please check https://github.com/kubernetes/dashboard/wiki/Code-conventions.

maciaszczykm commented on src/app/frontend/common/components/middleellipsis/component.js:
```diff
@@ -19,10 +19,26 @@ export default class MiddleEllipsisController {
+
+  isTruncated() {
+    let c = this.element_;
+    c.addClass("kd-middleellipsis_width");
```
Do not use `_` in here. Switch to `-`.

maciaszczykm commented:
https://github.com/kubernetes/dashboard/pull/2330#discussion_r140028573 and https://github.com/kubernetes/dashboard/pull/2330#discussion_r140028432 are not solved yet.

maciaszczykm commented:
Closing as stale.
